src/api-service/__app__/onefuzzlib/proxy.py

```diff
@@ -165,7 +165,10 @@ def stopping(self) -> None:
             self.stopped()
 
     def stopped(self) -> None:
+        if self.state != VmState.stopped:
+            self.set_state(VmState.stopped)
         logging.info(PROXY_LOG_PREFIX + "removing proxy: %s", self.region)
+        send_event(EventProxyDeleted(region=self.region, proxy_id=self.proxy_id))
```

Comment:
Should we only send this _after_ successfully calling `delete()`?

Reply:
I think it's a toss-up.  I was trying to remove the conceptual use-after-delete.

Reply:
That's legit. This and the other (related) comments about events / states are more discussion points. We can revisit the big picture going forward; I think the only thing that matters is that we have a well defined point in the business logic at which we expect to fire a given event!